File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

  int height;
  do {
    height = get_int("n: ");
  } while (height < 1 || height > 8);

  for (int i = 1; i <= height; i++) {

    // spaces
    for (int j = 1; j <= height - i; j++) {
      printf(" ");
    }

    // left ladder
    for (int k = 1; k <= i; k++) {
      printf("#");
    }

    // ladder delimeter
    printf("  ");

    // right ladder
    for (int k = 1; k <= i; k++) {
      printf("#");
    }

    // level delimeter
    printf("\n");
  }

  return 0;
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    while True:
        height = get_int('Height: ')
        if height > 0 and height < 9:
            break
        else:
            print('Height must be positive')

    ladder(height)


def ladder(rows):
    for row in range(1, rows + 1):
        # spaces
        print(' ' * (rows - row), end='')

        # left ladder
        print('#' * row, end='')

        # delim
        print('  ', end='')

        # right ladder
        print('#' * row)


if __name__ == "__main__":
    main()
